include/gatekeeper_sol.h

```diff
@@ -83,26 +83,39 @@ struct req_queue {
 	uint64_t         time_cpu_cycles;
 };
 
+/* Structures for each SOL instance. */
+struct sol_instance {
+	/*
+	 * Ring into which GK instances enqueue request packets
+	 * to be serviced and sent out by the Solicitor.
+	 */
+	struct rte_ring  *ring;
+
+	/* TX queue on the back interface. */
+	uint16_t         tx_queue_back;
+
+	/* Priority queue for request packets. */
+	struct req_queue req_queue;
+} __rte_cache_aligned;
+
 /* Configuration for the Solicitor functional block. */
 struct sol_config {
-	unsigned int       lcore_id;
-
 	/* Maximum number of requests to store in priority queue at once. */
-	unsigned int       pri_req_max_len;
+	unsigned int        pri_req_max_len;
```

Comment:
This field is used in a number of places and, with multiple SOL blocks, its meaning has changed in some of those places. If we keep the code as it is now, we have the following bug: during an attack like the evaluation that we've been using, the ring of the SOL blocks may be able to hold more packets than the mbuf pools of the GK blocks have. In this case, GK blocks won't have mbufs ready for new packets until SOL blocks release them. Therefore, we need to divide this parameter per the number of SOL instances.

Given that this field is used in multiple places to address slightly different needs, it is better to rename it to get help from the compiler to identify all the places that need to be reviewed. Given that we may want to preserve the original name, just add the suffix `__`, so we can easily remove this suffix later.

Implement this comment first as a new patch in this pull request to simplify the review of it. Once we are happy with this new patch, we may decide to merge it into the current patch for coherence.

Reply:
Thank you for creating the new patch, it has helped a lot to address this item.